### PR TITLE
Show errors from launching and fetching Metals

### PR DESCRIPTION
**Pull request by tgodzik**
Not sure if it's the right way to do it in typescript, but it would be good to have more information when something fails. Currently, we can only guess what happened. This should allow the user to at least paste us the error message.

I recommend using `Hide whitespace changes` option.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -232,7 +232,14 @@ function fetchAndLaunchMetals(context: ExtensionContext, javaHome: string) {
         javaConfig
       );
     },
-    () => {
+    (reason) => {
+      if (reason instanceof Error) {
+        const err = reason as Error;
```
Comment (gabro) [inline src/extension.ts:237]: No need for casting, if you already used `instanceof` (flow typing should understand that)

File: src/extension.ts [hunk 2]
```diff
@@ -361,523 +368,532 @@ function launchMetals(
...
+            editor.setDecorations(worksheetDecorationType, options);
+          else editor.setDecorations(decorationType, options);
+        } else {
+          outputChannel.appendLine(
+            `Ignoring decorations for non-active document '${params.uri}'.`
+          );
+        }
+      });
+    },
+    (err: Error) => {
```
Comment (gabro) [inline src/extension.ts:892]: if `err` is already of type `Error`, then no need to add the type here:

```suggestion
    (err) => {
```

If not, then we need to check it at runtime (like above)

**Review by gabro — CHANGES_REQUESTED** [review]
Couple of suggestions

File: src/extension.ts (hunk 1 above)
Comment (tgodzik) [reply src/extension.ts]: That's neat, fixed!

File: src/extension.ts (hunk 2 above)
Comment (tgodzik) [reply src/extension.ts]: Ach, right. I added a check.

**Review by gabro — APPROVED** [review]

**Pull request merged**